HotelBooking/HotelBooking.App/OperationsProcessing/OperationsManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Linq;
+using HotelBooking.Domain.Models;
+using HotelBooking.App.Operations.ExternalSystems;
+using HotelBooking.App.DataObjects;
+using HotelBooking.App.Enums;
+
+namespace HotelBooking.App.OperationsProcessing
+{
+    public class OperationsManager : IOperationsManager
+    {
+        private IOperationsProvider _operationsProvider;
+        private IExternalEmailSystem _emailSender;
+
+        public OperationsManager(IOperationsProvider operationsProvider, IExternalEmailSystem emailSender)
+        {
+            _operationsProvider = operationsProvider;
+            _emailSender = emailSender;
+        }
+
+        public BookingResult ProcessOperations(Reservation reservation)
+        {
+            var orderedOperations = _operationsProvider.GetOrderedOperations(reservation.Hotel);
+            var bookingResult = new BookingResult();
+
+            if (_operationsProvider.ContainsAllRequiredOperations(orderedOperations))
+            {
+                foreach (var operation in orderedOperations)
+                {
+                    operation.Process(reservation, bookingResult);
+
+                    if (operation.OperationResult.ShouldAbortBookingProcess)

```

Comment:
to jest 3 zagnizdzenie w tej metodzie co moze czynnic ja delikatnie mniej czytelna. Moja rada wychodz z metod tak szybko jak to mozliwe.

``` java
 var orderedOperations = _operationsProvider.GetOrderedOperations(reservation.Hotel);

            if (!_operationsProvider.ContainsAllRequiredOperations(orderedOperations))
            {
                return new BookingResult();
            }

            var bookingResult = new BookingResult();
            foreach (var operation in orderedOperations)
            {
                operation.Process(reservation, bookingResult);

                if (operation.OperationResult.ShouldAbortBookingProcess)
                {
                    AbortProcess(reservation, operation.OperationName);
                    break;
                }
            }

            DeduceOverallBookingResult(bookingResult);

            return bookingResult;
```